snapshot_dbg_cli_tests/test_init_command.py

```diff
@@ -0,0 +1,495 @@
+        return_value=f'{TEST_PROJECT_ID}-cdbg')
+    self.cli_services.get_firebase_default_rtdb_id = MagicMock(
+        return_value=f'{TEST_PROJECT_ID}-default-rtdb')
+
+    # Install a default happy path set of responses to mimic a project that is
+    # already fully configured to minimize duplicating uninteresting parts of
+    # the tests. The tests will override these defaults when appropriate.
+    self.gcloud_service_mock.is_api_enabled = MagicMock(return_value=True)
+
+    self.firebase_management_service_mock.project_get = MagicMock(
+        return_value=FirebaseProjectGetResponse(FirebaseProjectStatus.ENABLED,
+                                                build_firebase_project()))
+
+    self.firebase_management_service_mock.rtdb_instance_get = MagicMock(
+        return_value=DatabaseGetResponse(
+            status=DatabaseGetStatus.EXISTS,
+            database_instance=build_database_instance(TEST_PROJECT_ID,
+                                                      'USER_DATABASE')))
+
+    self.rtdb_service_mock.get_schema_version = MagicMock(return_value='1')
+
+  def run_cmd(self, testargs, expected_exception=None):
+    args = ['cli-test', 'init'] + testargs
+
+    # We patch the sleep to speed the tests up as the init command has a sleep,
+    # which slows the tests down.
+    with patch.object(sys, 'argv', args), \
+         patch('time.sleep', return_value=None), \
+         patch('sys.stdout', new_callable=StringIO) as out, \
+         patch('sys.stderr', new_callable=StringIO) as err:
+      if expected_exception is not None:
+        with self.assertRaises(expected_exception):
+          cli_run.run(self.cli_services)
+      else:
+        cli_run.run(self.cli_services)
+
+    return out, err
+
+  def test_only_default_location_allowed(self):
+    testargs = ['--location=us-west1']
+
+    out, err = self.run_cmd(testargs, expected_exception=SilentlyExitError)
+
+    self.assertEqual(
+        "ERROR: Currently the only supported location is 'us-central1'\n",
+        err.getvalue())
+    self.assertEqual('', out.getvalue())
+
+    # Ensure the test had not progressed at all, that check should be done
+    # first.
+    self.permissions_service_mock.check_required_permissions.assert_not_called()
+    self.gcloud_service_mock.is_api_enabled.assert_not_called()
+    self.firebase_management_service_mock.project_get.assert_not_called()
+    self.firebase_management_service_mock.rtdb_instance_get.assert_not_called()
+    self.cli_services.get_snapshot_debugger_rtdb_service.assert_not_called()
+    self.rtdb_service_mock.get_schema_version.assert_not_called()
+
+  def test_permissions_check_done_as_expected(self):
+    testargs = []
+
+    self.permissions_service_mock.check_required_permissions = MagicMock(
+        side_effect=SilentlyExitError())
+
+    self.run_cmd(testargs, expected_exception=SilentlyExitError)
+
+    self.permissions_service_mock.check_required_permissions.assert_called_once(
+    )
```

Comment:
The opening '(' is at the 80 char mark and so that's how yapf formats it.

An alternative would be to replace it with something like:
```
    check_perms = self.permissions_service_mock.check_required_permissions
    check_perms.assert_called_once()
```

Thoughts?

Reply:
Per offline discussion, will simply leave it as is.